pkg/disttask/framework/taskexecutor/manager.go

```diff
@@ -196,7 +204,14 @@ func (m *Manager) onRunnableTasks(tasks []*proto.Task) {
 		return
 	}
 	tasks = m.filterAlreadyHandlingTasks(tasks)
+
+	priorityQueue := make(proto.TaskPriorityQueue, 0)
+	heap.Init(&priorityQueue)
 	for _, task := range tasks {
+		heap.Push(&priorityQueue, proto.WrapPriorityQueue(task))
+	}
```

Comment:
Not considering handling tasks

Reply:
will be filtered

Reply:
> has been filtered

But we will take running tasks into consideration.

Reply:
> > has been filtered
> 
> But we should take running tasks into consideration.

Reply:
Do you mean after TiDB restarts?